src/scripts/charts/BaseChart.js

```diff
@@ -38,7 +38,11 @@ export default class BaseChart {
 		this.has_legend = has_legend;
 
 		this.colors = colors;
-		if(!this.colors || (this.data.labels && this.colors.length < this.data.labels.length)) {
+		const list = type === 'percentage' || type === 'pie'
+			? this.data.labels
+			: this.data.datasets;
+
+		if(!this.colors || (list && this.colors.length < list.length)) {
 			this.colors = ['light-blue', 'blue', 'violet', 'red', 'orange',
 				'yellow', 'green', 'light-green', 'purple', 'magenta'];
 		}
```

Comment:
You'll probably want to replace these with hex-codes and test stuff out again.

Reply:
Do you think thats needed? They currently get mapped to the correct hex codes using the color_map field in utils/colors.js, rather than the css. I did this to allow for backwards compatibility for anyone currently using the color names, which also allowed for the hex codes to be kept in one place. Or am I misunderstanding you?

Reply:
Sure missed that.

Reply:
However, it looks like there's more calls to `get_color()` than necessary. You see, currently it is called at every place where and when the color is actually set (while setting the style). But because the colors are already received beforehand in the constructor, it would be much more efficient to do the conversions there itself. That way we'd have the real colors ready for use whenever needed (without remembering to call `get_color()`: There is a case in `make_gradient()` below which demonstrates this, where while setting the `stop-color` we seemingly forgot a `get_color()` call? ;) ).

So in this light, it makes sense to store the defaults in the constructor as hex codes itself in case no colors are passed, while in case they are, going in, using `get_color()` for the color for each dataset in case of axis, while doing the same for the colors array for pies.